src/pages/ScholarshipForm.jsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState } from 'react';
+import { Container, Button, TextField } from '@material-ui/core';
+
+function ScholarshipForm() {
+  const [formFieldStates, setFormFieldStates] = useState({
+    name: '',
+    deadline: new Date(),
+    description: '',
+    amountType: '',
+    amount: 0,
+    minAmount: 0,
+    maxAmount: 0,
+    website: '',
+    scholarshipType: '',
+    active: false,
+  });
+
+  function updateFn(id) {
+    return (e) =>
+      setFormFieldStates({ ...formFieldStates, [id]: e.target.value });
+  }
+
+  return (
+    <Container maxWidth="md">
+      <form>
+        <h1>Submit a Scholarship</h1>
+        <TextField
+          id="name"
+          label="Scholarship Name *"
+          value={formFieldStates.name}
+          onChange={updateFn('name')}
+        />
+        <TextField
+          id="description"
+          label="Description *"
+          value={formFieldStates.description}
+          onChange={updateFn('description')}
+        />
+        <TextField
+          id="website"
+          label="Website *"
+          value={formFieldStates.website}
+          onChange={updateFn('website')}
+        />
```

Comment:
optional: one slight improvement might be to have a helper function that returns a TextField with the `value` and `onChange` set based on the `id` so you can write this as:

```
{requiredTextField('website', 'Website')}
```

This would be different from a TextField function component class as it would be scoped within this function so that it can access state.

Reply:
Oo yeah that looks cleaner